docs/host-and-deploy/index.md

```diff
@@ -0,0 +1,175 @@
+
+Not all hosting scenarios permit the enforcement of rewrite rules. Additional information is provided in the hosting configuration examples that follow.
+
+## Host with ASP.NET Core
+
+ASP.NET Core apps that host Blazor apps configure the server-side app's middleware pipeline with the `UseBlazor` extension method on [IApplicationBuilder](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.builder.iapplicationbuilder) in `Startup.Configure`:
+
+```csharp
+public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+{
+    if (env.IsDevelopment())
+    {
+        app.UseDeveloperExceptionPage();
+    }
+
+    app.UseBlazor<Client.Program>();
+}
+```
+
+The `UseBlazor` extension method performs the following tasks:
+
+* Reads the Blazor configuration file (*.blazor.config*) from the `ClientAssemblyPath`, which is either an option provided to the `UseBlazor` extension method or defaulted to the app's assembly file location.
+* Configures [Static File Middleware](https://docs.microsoft.com/aspnet/core/fundamentals/static-files) to serve Blazor's static assets from the *dist* folder. In the Development environment, the files in *wwwroot* are served.
+* Configure live reloading when in the Development environment and Debug configuration.
+* Configure single-page application routing for files outside of the *_framework* folder, which serves the default page (*index.html*) for any request that hasn't been served by a prior Static File Middleware instance.
+
+For general information on ASP.NET Core app hosting and deployment, see:
+
+* [Host and deploy ASP.NET Core](https://docs.microsoft.com/aspnet/core/host-and-deploy)
+* [Host ASP.NET Core on Azure App Service](https://docs.microsoft.com/aspnet/core/host-and-deploy/azure-apps/)
+* [.NET Core application deployment](https://docs.microsoft.com/dotnet/core/deploying/)
+* [Deploying .NET Core apps with command-line interface (CLI) tools](https://docs.microsoft.com/dotnet/core/deploying/deploy-with-cli)
+* [Deploying .NET Core apps with Visual Studio](https://docs.microsoft.com/dotnet/core/deploying/deploy-with-vs)
+
+## Docker/Linux hosting
+
+The first line of the Blazor config file (*.blazor.config*) configures the source MSBuild path to `.`. A consequence of this setting is that the webroot path is set to `.\wwwroot`, which results in published files being split between *wwwroot* and *dist* folders. For an unsupported workaround, see [On linux UseBlazor extension method fails with "The path must be absolute"](https://github.com/aspnet/Blazor/issues/376).
+
+## Static deployment strategies
+
+The following static deployment strategies make the Blazor deployment's static assets available directly for download. Conceptually, other static hosting system configurations are similar.
+
+### IIS
```

Comment:
I wonder how useful it is really to describe how to configure IIS to serve static content. Might there be some external IIS documentation we could link to instead? It's not Blazor-specific. What do you think @danroth27?

If we were going to put in detailed instructions, I'd suggest focusing on Azure app service deployment would be a much higher priority.

Reply:
@rstropek says he has direct experience with Blazor client-server apps on Azure App Service, and he can link our aspnet/Docs content to help explain and handle troubleshooting. We have an issue open to cover it: https://github.com/aspnet/Blazor.Docs/issues/18

RE: IIS static file server

I looked ... the coverage isn't as clear and focused as I would've liked. My (biased) impression is that following these instructions makes IIS config simple and fast. I'll check their IIS topic again and report back.

Reply:
Ugh .....

* https://docs.microsoft.com/iis/manage/creating-websites/scenario-build-a-static-website-on-iis
* https://docs.microsoft.com/iis/web-hosting/getting-started/deploying-a-static-content-server

Yeah ... that's what I saw a few days ago. I'd prefer NOT to link those for several reasons, including missing info and references to WS1012.

Reply:
I would prefer to not document how to install and setup a site with IIS. That should be documented in the IIS docs. I do however agree that piecing together the full set of steps listed here from the IIS docs is not easy. @shirhatti You should consider working with @guardrex to make the IIS docs better so he doesn't have to keep documenting all these steps 😃 

I do think we need to document that we generate a web.config file with the right mime type mappings and URL rewrite rules, and that you nee to install the URL rewrite module.

Reply:
I also think documenting Azure App Service is required. At a minimum we should document that from VS you can right-click -> Publish.